packages/token/src/index.ts

```diff
@@ -27,25 +32,39 @@ export const STABLECOINS = ["USDC", "USDT"];
 
 /** The token class */
 export class Token {
-    private static tokenMapping: TokenMapping;
+    private static tokenMappings: Partial<Record<ChainId | "default", TokenMapping>>;
 
     /** Fetch a remap from the repo */
     static async fetchRemapFromRepo(chain: number) {
-        const name = getSDKConfig(chain).chainSpecifier;
-        const url = new URL(`${REMAP_BASE_URL}${name}.json`);
+        if (!isSupportedChainId(chain)) {
+            throw new Error(`Chain ${chain} is not supported`);
+        }
+
+        const chainSpecifier = CHAIN_SPECIFIERS[chain];
+        const url = new URL(`${REMAP_BASE_URL}${chainSpecifier}.json`);
         const res = await fetch(url);
         const data = (await res.json()) as TokenMapping;
-        Token.processRemap(data.tokens);
+        Token.addRemap(chain, data.tokens);
     }
 
-    /** Parse a remap from a stringified JSON object */
+    /**
+     * Parse a remap from a stringified JSON object.
+     *
+     * @deprecated Use `Token.addRemapFromString` instead.
```

Comment:
Do we need to deprecate? This is only used internally so we can just migrate all callsites over to the new method.

Reply:
It's still a public method on the class, so I'd want to mark deprecated before deleting.